CIP-????/README.md

```diff
@@ -0,0 +1,250 @@
+
+In summary, the goals are:
+
+1. [x] to make some first experimental baby steps in the realm of governance;
+1. [x] to capture participation data from SPOs;
+1. [x] to raise awareness amongst SPOs regarding their future role in governance;
+1. [x] to keep the Voltaire dynamics up in the ecosystem while other efforts are being pursued;
+1. [x] to improve relations between the Cardano Foundation & SPOs for better mutual understanding and fruitful conversations.
+
+
+## Specification
+
+### Overview
+
+Polls will be multiple-choice questions by The Cardano Foundation with pre-defined answers to choose from.
+
+Here's an example of a question and answers:
+
+- _Pineapples on pizza?_
+  - [ ] yes
+  - [ ] no
+
+The serialised question and answers will be posted on-chain and signed by one of the delegate genesis keys owned by The Cardano Foundation. Answers will be provided on-chain by participating SPOs via transaction metadata referring to:
+
+- The question and answers
+- The index of the chosen answer from the available choices
+- A signature from the SPO's current VRF credentials or Ed25519 declared cold credentials.
+
+> **Note**
+> In this document, every time we refer to a _serialized object_, we refer to its **canonical** CBOR representation. In particular, keys in a map are always ordered alphabetically.
+
+### Question structure
+
+A question is posted in a transaction's metadata using the metadata label `94` and the following metadata structure:
+
+```cbor
+question =
+  { 0: prompt
+  , 1: [ * choice ]
+  , ? "_": nonce
+  }
+
+prompt =
+  text .size (0..64) / [ * text .size (0..64) ]
+
+choice =
+  text .size (0..64) / [ * text .size (0..64) ]
+
+nonce =
+  uint
+```
+
+A nonce is optionally included to provide non-replayability should the same question and answers be asked multiple times over different periods. The transaction carrying a question **must** be signed by one of the genesis delegate keys to be considered valid. This genesis key signature isn't captured in the metadata but in the transaction itself as an extra signatory.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{ 94:
+  { 0: "Pineapples on pizza?"
+  , 1:
+    [ "yes"
+    , "no"
+    ]
+  }
+}
+```
+
+Which serialises down to:
+
+```
+A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573
+626E6F
+```
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
```

Comment:
```suggestion
```cbor
answer =
  { 2: question_hash
  , 3: choice
  , ? 4: vrf_witness
  , ? 5: ed25519_witness
  }

question_hash =
  bytes .size 32

vrf_witness =
  [ $vrf_vkey, vrf_proof ]

vrf_proof =
  [ bytes .size 64, bytes .size 16 ]

ed25519_witness =
  [ $vkey, $signature ]
  
ballot = [ + answer ]
```
```

Here a `ballot` can be viewed as a simple array of question answers. Since each answer contains the hash of the question it answers, users may answer one (or more) of a multi-question poll in the same data structure.